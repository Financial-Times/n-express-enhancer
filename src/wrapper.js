import { toMiddleware, toMiddlewares } from './convertor';

export const toMiddlewareWrapper = input => {
	switch (typeof input) {
		case 'function':
			return toMiddleware(input);
		case 'object':
			Object.keys(input).forEach(key => {
				if (typeof input[key] !== 'function') {
					throw Error(
						'all methods in an object of operation function bundle need to be valid operation function',
					);
				}
			});
			return toMiddlewares(input);
		default:
			throw Error(
				'input of toMiddleware() needs to be an operation function or a set of operation functions wrapped in an object',
			);
	}
};

export default toMiddlewareWrapper;
