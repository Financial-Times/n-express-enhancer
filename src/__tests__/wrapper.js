import { toMiddlewareWrapper } from '../wrapper';

describe('toMiddlewareWrapper', () => {
	it('enhance input function with toMiddleware', () => {
		const operationFunction = () => {};
		const enhanced = toMiddlewareWrapper(operationFunction);
		expect(typeof enhanced).toBe('function');
		expect(enhanced.name).toBe('operationFunction');
	});

	it('enhance input function bundle with toMiddlewares', () => {
		const createOperationFunction = () => () => {};
		const bundle = {
			methodA: createOperationFunction(),
			methodB: createOperationFunction(),
		};
		const enhanced = toMiddlewareWrapper(bundle);
		expect(typeof enhanced).toBe('object');
		expect(typeof enhanced.methodA).toBe('function');
		expect(enhanced.methodA.name).toBe('methodA');
		expect(typeof enhanced.methodB).toBe('function');
		expect(enhanced.methodB.name).toBe('methodB');
	});

	it('throws error if given invalid input', () => {
		const inputString = () => toMiddlewareWrapper('test');
		expect(inputString).toThrowErrorMatchingSnapshot();

		const inputInvalidObject = () =>
			toMiddlewareWrapper({
				foo: 'bar',
			});
		expect(inputInvalidObject).toThrowErrorMatchingSnapshot();
	});
});
