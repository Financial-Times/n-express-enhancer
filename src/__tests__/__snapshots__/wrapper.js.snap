// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toMiddlewareWrapper throws error if given invalid input 1`] = `"input of toMiddleware() needs to be an operation function or a set of operation functions wrapped in an object"`;

exports[`toMiddlewareWrapper throws error if given invalid input 2`] = `"all methods in an object of operation function bundle need to be valid operation function"`;
